File: LeetCode/TwoSum.py
def twoSum(arr, target):
	map = {}
	for i in range(len(arr)):
		temp = target - arr[i]
		if temp in map:
			return [map[temp], i]
		else:
			map[arr[i]] = i


arr = list(map(int, input().split()))
target = int(input())
res = twoSum(arr, target)
print(res)

File: LeetCode/TwoSum/TwoSum.ts
function twoSum(nums: number[], target: number): number[] {
  let obj: object = {};
  for (let i = 0; i < nums.length; i++) {
    let temp = target - nums[i];
    if (obj[temp] !== undefined) {
      return [obj[temp], i];
    }
    obj[nums[i]] = i;
  }

  return [-1, -1];
}
